argus/backend/service/results_service.py

```diff
@@ -250,21 +297,64 @@ def calculate_graph_ticks(graphs: List[Dict]) -> dict[str, str]:
     return {"min": min_x[:10], "max": max_x[:10]}
 
 
+def _identify_most_changed_package(packages_list: list[PackageVersion]) -> str:
+    version_date_changes: dict[str, set[tuple[str, str]]] = defaultdict(set)
+
+    for package_version in packages_list:
+        version_date_changes[package_version.name].add((package_version.version, package_version.date))
+
+    return max(version_date_changes, key=lambda k: len(version_date_changes[k]))
+
+
+def _split_results_by_release(packages: dict[str, list[PackageVersion]], main_package: str) -> ReleasesMap:
+    releases_map = defaultdict(list)
+    for run_id, package_versions in packages.items():
+        for package in package_versions:
+            if package.name == main_package:
+                if "dev" in package.version:
+                    major_version = 'dev'
+                else:
+                    major_version = '.'.join(package.version.split('.')[:2])
+                releases_map[major_version].append(run_id)
+    return releases_map
+
+
+def create_chartjs(table: ArgusGenericResultMetadata, data: list[ArgusGenericResultData], best_results: dict[str, List[BestResult]],
+                   releases_map: ReleasesMap) -> List[Dict]:
+    """
+    Create Chart.js-compatible graph for each column in the table.
+    """
+    graphs = []
+    columns = [column for column in table.columns_meta if column.type != "TEXT"]
+
+    for column in columns:
+        datasets = create_datasets_for_column(table, data, best_results, releases_map, column)
+
+        if datasets:
+            min_y, max_y = get_min_max_y(datasets)
+            datasets = coerce_values_to_axis_boundries(datasets, min_y, max_y)
+            options = create_chart_options(table, column, min_y, max_y)
+            graphs.append({"options": options, "data": {"datasets": datasets}})
+
+    return graphs
+
+
 class ResultsService:
 
     def __init__(self):
         self.cluster = ScyllaCluster.get()
 
     @cache
-    def _ignored_results(self, test_id: UUID) -> list[str]:
+    def _get_runs_details(self, test_id: UUID) -> RunsDetails:
         plugin_query = self.cluster.prepare("SELECT id, plugin_name FROM argus_test_v2 WHERE id = ?")
         plugin_name = self.cluster.session.execute(plugin_query, parameters=(test_id,)).one()['plugin_name']
         plugin = TestRunService().get_plugin(plugin_name)
-        ignored_runs_query = self.cluster.prepare(f"SELECT id, investigation_status FROM {plugin.model.table_name()} WHERE test_id = ?")
-        ignored_runs = [run["id"] for run in self.cluster.session.execute(ignored_runs_query, parameters=(test_id,)).all()
-                        if run["investigation_status"].lower() == "ignored"]
-        LOGGER.debug(f"Ignored runs for test {test_id}: {ignored_runs}")
-        return ignored_runs
+        ignored_runs_query = self.cluster.prepare(
+            f"SELECT id, investigation_status, packages FROM {plugin.model.table_name()} WHERE test_id = ?")
```

Comment:
CQLengine supports sparse queries, like so:

```python
model.filter(test_id=test_id).only(["id", "investigation_status", "packages"]).all()
```

I haven't really been using them but in this case they are perfect (as long as you don't need asyncio).

Reply:
I use it mostly to reuse 'prepared' queries, can we do it with your proposed solution?

Reply:
Unfortunately CQLEngine seems to always do a `SimpleStatement`, so you're trading some potential performance for convenience.

Reply:
I think convenience is not worth it - I don't find it much more complex than model method.